src/Memory/include/Memory/MemoryInfoProducer.h

```diff
@@ -0,0 +1,45 @@
+// Copyright (c) 2021 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef ORBIT_MEMORY_INFO_PRODUCER_H_
+#define ORBIT_MEMORY_INFO_PRODUCER_H_
+
+#include <atomic>
+#include <thread>
+
+#include "Memory/MemoryInfoListener.h"
+#include "capture.pb.h"
+
+namespace orbit_memory {
+class MemoryInfoProducer {
+ public:
+  explicit MemoryInfoProducer(const orbit_grpc_protos::CaptureOptions& capture_options)
+      : collect_memory_info_(capture_options.collect_memory_info()),
+        sampling_period_ns_(capture_options.memory_sampling_period_ns()) {}
+
+  MemoryInfoProducer(const MemoryInfoProducer&) = delete;
+  MemoryInfoProducer& operator=(const MemoryInfoProducer&) = delete;
+  MemoryInfoProducer(MemoryInfoProducer&&) = delete;
+  MemoryInfoProducer& operator=(MemoryInfoProducer&&) = delete;
+
+  ~MemoryInfoProducer() { Stop(); }
+
+  void SetListener(MemoryInfoListener* listener) { listener_ = listener; }
+
+  void Start();
+  void Stop();
+
+ private:
+  void Run();
+  void ProduceTotalMemoryUsageInfoAndSendToListener();
+
+  bool collect_memory_info_;
```

Comment:
maybe avoid starting this producer if this option is set to false instead of checking it when we start the thread?

Reply:
Done. Now I let the `MemoryInfoHandler` to check `CaptureOptions::collect_memory_info_` and decide whether to start the producer. Thanks!